### ISSUE TITLE
Fit convolution parameters to experimental data

### ISSUE THREAD

**Issue by maurov** [xraypy/xraylarch#436]
@newville @mretegan

This is a follow-up of #435. I was wondering if we could make a fitting routine (based on lmfit) that minimizes the convolutions parameters for a FDMNES simulation in order to best fit the experimental data. Usually this is a manual/visual procedure the user does or, in the worst case, is not done at all and default convolution parameters are simply kept.

If you have ideas or you are willing to share some code, please, feel free to post here.

**Comment by newville** [comment]
@maurov Yes, thanks for #435   This sort of fitting of the broadening is exactly what I want, too.   I find myself (and sometimes see others) spending a lot of time twiddling these parameters.

It is OK to make this be a part of XAS Viewer / Larix ?  Like, read in and somehow present unconvolved FDMNES calculations (I would think that "as an XAS Viewer group" would be OK to start),  and the convolve to best match experimental data?

**Comment by maurov** [comment]
@newville 

> It is OK to make this be a part of XAS Viewer / Larix ? Like, read in and somehow present unconvolved FDMNES calculations (I would think that "as an XAS Viewer group" would be OK to start), and the convolve to best match experimental data?

Yes, this would be great! If you have in mind how should be a function that given unconvoluted XANES calculations - FDMNES or other read as a mu(E) - plus experimental one get the best convolution parameters automagically, it would be wonderful.

**Comment by maurov** [comment]
@newville @mretegan, FYI:

I have found that in section E of the [FDMNES manual](https://cloud.neel.cnrs.fr/index.php/s/nL2c6kH2PLwcB5r), page 71, the possibility to fit the convolution parameters to the experimental spectrum. I have highlighted what I think are the main parameters that affects the convolution in first order.

![image](https://github.com/xraypy/xraylarch/assets/545294/504aa74a-97af-4916-a575-61a025572fe7)

By experience, it is also important how one goes from `gamma_hole` to `gamma_max`, that is, going from an arctangent-like to an erf-like function, but this means 3 more parameters in the fit.

At the moment I do not have time to implement this in lmfit, but if any of you is planning to do it, please, keep posted here.
